ReactTestApp-DevSupport.podspec

```
@@ -6,12 +6,12 @@ require 'json'
 deployment_target = lambda do |target_platform|
   xcodeproj = File.join(__dir__, target_platform, 'ReactTestApp.xcodeproj')
   project = Xcodeproj::Project.open(xcodeproj)
-  setting = case target_platform
-            when 'ios'
-              'IPHONEOS_DEPLOYMENT_TARGET'
-            when 'macos'
-              'MACOSX_DEPLOYMENT_TARGET'
-            end
+  settings = {
+    'ios' => 'IPHONEOS_DEPLOYMENT_TARGET',
+    'macos' => 'MACOSX_DEPLOYMENT_TARGET',
+    'visionos' => 'XROS_DEPLOYMENT_TARGET',
+  }
```

We should use the newer JSON-style syntax:
```suggestion
  settings = {
    ios: 'IPHONEOS_DEPLOYMENT_TARGET',
    macos: 'MACOSX_DEPLOYMENT_TARGET',
    visionos: 'XROS_DEPLOYMENT_TARGET',
  }
```
This will make all keys a Symbol, so we will also have to change the callsite as well:
```ruby
  s.ios.deployment_target = deployment_target.call(:ios)
  s.osx.deployment_target = deployment_target.call(:macos)
  s.visionos.deployment_target = deployment_target.call(:visionos)
```

---

Doesn't work with line 7;
> [!] Invalid `ReactTestApp-DevSupport.podspec` file: no implicit conversion of Symbol into String.
> xcodeproj = File.join(__dir__, **target_platform**, 'ReactTestApp.xcodeproj')